packages/debug/src/browser/debug-session.tsx

```diff
@@ -274,19 +287,26 @@ export class DebugSession implements CompositeTreeElement {
     }
 
     protected async initialize(): Promise<void> {
-        const response = await this.connection.sendRequest('initialize', {
-            clientID: 'Theia',
-            clientName: 'Theia IDE',
-            adapterID: this.configuration.type,
-            locale: 'en-US',
-            linesStartAt1: true,
-            columnsStartAt1: true,
-            pathFormat: 'path',
-            supportsVariableType: false,
-            supportsVariablePaging: false,
-            supportsRunInTerminalRequest: true
-        });
-        this.updateCapabilities(response?.body || {});
+        const clientName = FrontendApplicationConfigProvider.get().applicationName;
+        try {
+            const response = await this.connection.sendRequest('initialize', {
+                clientID: clientName.toLocaleLowerCase().replace(/\s+/g, '_'),
```

Comment:
Is there a technical reason to use the application name here or is this just branding? I feel the same _implementation_ should have the same client id, kind of like the `user-agent` of a browser.

Reply:
https://github.com/eclipse-theia/theia/pull/11984/commits/57e7f39edcb2f5f3236d26552d2c303594b819e4